plugin.py

```diff
@@ -250,11 +251,36 @@ def install_or_update(cls) -> None:
             raise
 
     def on_pre_server_command(self, command: Mapping[str, Any], done_callback: Callable[[], None]) -> bool:
-        if command["command"] in ("rust-analyzer.runSingle", "rust-analyzer.runDebug"):
-            open_runnables_in_terminus(sublime.active_window(), command["arguments"])
-            done_callback()
+        command_name = command["command"]
+        try:
+            session = self.weaksession()
+            if not session:
+                return False
+            if command_name in ("rust-analyzer.runSingle", "rust-analyzer.runDebug"):
+                open_runnables_in_terminus(sublime.active_window(), command["arguments"])
+                done_callback()
+                return True
+            elif command_name == "rust-analyzer.showReferences":
+                return self._handle_show_references(session, command, done_callback)
```

Comment:
Would call `done_callback` in this scope for consistency and because feels a bit better semantically (if we treat `_handle_show_references` as something that can be called from different contexts).